### PR TITLE
feat: add readme for the repo with policy details

### LINKED ISSUE #4: Add readme for the repo with relevant details on each policies

### PR DESCRIPTION
**Pull request by sakshi-1505**
fixes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,32 @@
...
+
+### Binary Artifacts Policy
+- Prevents binary artifacts from being committed to the repositories.
+- Ensures that source code is human-readable and free from hidden vulnerabilities.
+
+### Branch Protection Policy
+- Verifies that the main branches (e.g., `main`) have branch protection rules enforced, such as required reviews, status checks, and more.
+
+### SECURITY.md Policy
+- Checks that a security policy file named `SECURITY.md` is present and properly filled out in each repository. This file should detail how to report security vulnerabilities.
```
Comment (jpkrohling) [inline README.md:32]: Is this our current policy? I think we have an org-wide file that applies to repos that don't have an explicit security.md. What would happen in that case?

**Review by jpkrohling — COMMENTED** [review]
LGTM,  I have a question about the security.md rule though.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,32 @@
-# .allstar
-Enable and house Allstar policies centrally for the organizatio
+# Allstar Security Policy Enforcement
+
+## Overview
+
+This repository outlines the security policy enforcement for the OpenTelemetry organization, using the Allstar GitHub App. Allstar helps enforce security best practices by automatically checking and ensuring our repositories comply with our established policies.
```
Comment (codeboten) [inline README.md:5]:
```suggestion
This repository outlines the security policy enforcement for the OpenTelemetry organization, using the [Allstar GitHub App](https://github.com/ossf/allstar). Allstar helps enforce security best practices by automatically checking and ensuring our repositories comply with our established policies.
```

File: README.md (hunk 1 above)
Comment (codeboten) [reply README.md:32]: I'm assuming that allstar recognizes a `SECURITY.MD` file defined in the org, but it might be worth calling out specifically here

Comment (sakshi-1505) [reply README.md:32]: Here it is mentioned that the policy will check repo `security.md` https://github.com/ossf/allstar#securitymd BUT; I delved into the code so if the org has a org-wide file that propagate the attribute `IsSecurityEnabled` for repo object as true so Allstar will be happy https://github.com/ossf/allstar/blob/3521ed8096d5ddcb5c3f7c9804393e79fca3a6aa/pkg/policies/security/security.go#L137

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**